scripts/HPC_chatbot_preprocessor/chatbot_parser.py

```diff
@@ -0,0 +1,683 @@
+import json
+import os
+import re
+import shutil
+import yaml
+from itertools import chain
+from jinja2 import FileSystemLoader, Environment, ChoiceLoader
+
```

Comment:
this is a very very long and brittle script without unittests. you might want to add some minimal code to se we can make sure this stays working at least.